[2025-06-15T00:42:16.036+0100] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-15T00:42:16.042+0100] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: customer_churn_training_dag.train_on_full_dataset manual__2025-06-14T23:42:07.150431+00:00 [queued]>
[2025-06-15T00:42:16.044+0100] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: customer_churn_training_dag.train_on_full_dataset manual__2025-06-14T23:42:07.150431+00:00 [queued]>
[2025-06-15T00:42:16.044+0100] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-06-15T00:42:16.051+0100] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): train_on_full_dataset> on 2025-06-14 23:42:07.150431+00:00
[2025-06-15T00:42:16.053+0100] {standard_task_runner.py:64} INFO - Started process 221164 to run task
[2025-06-15T00:42:16.055+0100] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'customer_churn_training_dag', 'train_on_full_dataset', 'manual__2025-06-14T23:42:07.150431+00:00', '--job-id', '127', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpynso6h7i']
[2025-06-15T00:42:16.055+0100] {standard_task_runner.py:91} INFO - Job 127: Subtask train_on_full_dataset
[2025-06-15T00:42:16.074+0100] {task_command.py:426} INFO - Running <TaskInstance: customer_churn_training_dag.train_on_full_dataset manual__2025-06-14T23:42:07.150431+00:00 [running]> on host neonm.
[2025-06-15T00:42:16.104+0100] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='customer_churn_training_dag' AIRFLOW_CTX_TASK_ID='train_on_full_dataset' AIRFLOW_CTX_EXECUTION_DATE='2025-06-14T23:42:07.150431+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-14T23:42:07.150431+00:00'
[2025-06-15T00:42:16.105+0100] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-15T00:42:16.105+0100] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-06-15T00:42:16.105+0100] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n            cd /home/press/coding/acronis_interview_v02 &&             python scripts/model_training.py                 --data_path kaggle_datasets/telecom_customer_churn.csv                 --n_estimators 100                 --max_depth 10                 --top_n_features 20                 --mlflow_tracking_uri http://127.0.0.1:5000\n        ']
[2025-06-15T00:42:16.109+0100] {subprocess.py:86} INFO - Output:
[2025-06-15T00:42:17.084+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:17,083 - INFO - Loading data from kaggle_datasets/telecom_customer_churn.csv
[2025-06-15T00:42:17.120+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:17,120 - INFO - Converting integer columns to float to handle potential missing values...
[2025-06-15T00:42:17.141+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:17,141 - INFO - Logging training data...
[2025-06-15T00:42:18.369+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,369 - INFO - Training RandomForestClassifier...
[2025-06-15T00:42:18.836+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,835 - INFO - Accuracy: 0.8147622427253371
[2025-06-15T00:42:18.836+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,835 - INFO - AUC: 0.8819292670955076
[2025-06-15T00:42:18.836+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,835 - INFO - Precision: 0.864516129032258
[2025-06-15T00:42:18.836+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,835 - INFO - Recall: 0.3582887700534759
[2025-06-15T00:42:18.836+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,835 - INFO - F1 Score: 0.5066162570888468
[2025-06-15T00:42:18.945+0100] {subprocess.py:93} INFO - 2025-06-15 00:42:18,944 - INFO - Creating feature importance chart...
[2025-06-15T00:42:20.747+0100] {subprocess.py:93} INFO - Registered model 'random_forest_model' already exists. Creating a new version of this model...
[2025-06-15T00:42:20.761+0100] {subprocess.py:93} INFO - 2025/06/15 00:42:20 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: random_forest_model, version 49
[2025-06-15T00:42:20.761+0100] {subprocess.py:93} INFO - Created version '49' of model 'random_forest_model'.
[2025-06-15T00:42:20.789+0100] {subprocess.py:93} INFO - 0.8147622427253371
[2025-06-15T00:42:20.789+0100] {subprocess.py:93} INFO - üèÉ View run likeable-ant-729 at: http://127.0.0.1:5000/#/experiments/1/runs/51ff6534f5d44e14b506ced678c7415f
[2025-06-15T00:42:20.789+0100] {subprocess.py:93} INFO - üß™ View experiment at: http://127.0.0.1:5000/#/experiments/1
[2025-06-15T00:42:20.944+0100] {subprocess.py:97} INFO - Command exited with return code 0
[2025-06-15T00:42:20.945+0100] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-15T00:42:20.955+0100] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=customer_churn_training_dag, task_id=train_on_full_dataset, run_id=manual__2025-06-14T23:42:07.150431+00:00, execution_date=20250614T234207, start_date=20250614T234216, end_date=20250614T234220
[2025-06-15T00:42:20.970+0100] {local_task_job_runner.py:243} INFO - Task exited with return code 0
[2025-06-15T00:42:20.977+0100] {taskinstance.py:3503} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-06-15T00:42:20.980+0100] {local_task_job_runner.py:222} INFO - ::endgroup::
